src/core_editor/line_buffer.rs

```diff
@@ -179,6 +179,30 @@ impl LineBuffer {
             .unwrap_or_else(|| self.lines.len())
     }
 
+    /// Cursor position *at end of* the next word to the right
+    pub fn word_right_end_index(&self) -> usize {
+        self.lines[self.insertion_point..]
+            .split_word_bound_indices()
+            .filter_map(|(i, word)| {
+                word.grapheme_indices(true)
+                    .next_back()
+                    .map(|x| x.0)
+                    .filter(|k| *k != 0 && !is_whitespace_str(word))
+                    .map(|x| self.insertion_point + x + i)
+            })
+            .next()
+            .unwrap_or_else(|| self.lines.len().saturating_sub(1))
```

Comment:
The default case will still panic for a `ö` in the last character position like this.